AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/Connection/DataStoreConnectionScenario6Tests.swift

```diff
@@ -90,6 +90,119 @@ class DataStoreConnectionScenario6Tests: SyncEngineIntegrationTestBase {
+        var resultPost: Post6?
+        Amplify.DataStore.query(Comment6.self, byId: comment.id) { result in
+            switch result {
+            case .success(let queriedCommentOptional):
+                guard let queriedComment = queriedCommentOptional else {
+                    XCTFail("Could not get comment")
+                    return
+                }
+                XCTAssertEqual(queriedComment.id, comment.id)
+                resultPost = queriedComment.post
+                getCommentCompleted.fulfill()
+            case .failure(let response):
+                XCTFail("Failed with: \(response)")
+            }
+        }
+        wait(for: [getCommentCompleted], timeout: TestCommonConstants.networkTimeout)
+
+        guard let fetchedPost = resultPost else {
+            XCTFail("Could not get post")
+            return
+        }
+
+        guard let fetchedBlog = fetchedPost.blog else {
+            XCTFail("Could not get blog")
+            return
+        }
+
+        // check if post was eagerly loaded
+        XCTAssertEqual(fetchedPost.id, post.id)
+        XCTAssertEqual(fetchedPost.title, post.title)
+
+        // check if blog was eagerly loaded
+        XCTAssertEqual(fetchedBlog.id, blog.id)
+        XCTAssertEqual(fetchedBlog.name, blog.name)
+    }
+
+    func testGetPostThenFetchBlogAndComment() throws {
+        try startAmplifyAndWaitForSync()
+        guard let blog = saveBlog(name: "name"),
+              let post = savePost(title: "title", blog: blog),
+              let comment = saveComment(post: post, content: "content") else {
+            XCTFail("Could not create blog, post, and comment")
+            return
+        }
+
+        // fetch post
+        let getPostCompleted = expectation(description: "get post complete")
+        var resultComments: List<Comment6>?
+        var resultBlog: Blog6?
+        Amplify.DataStore.query(Post6.self, byId: post.id) { result in
+            switch result {
+            case .success(let queriedPostOptional):
+                guard let queriedPost = queriedPostOptional else {
+                    XCTFail("Could not get comment")
+                    return
+                }
+                XCTAssertEqual(queriedPost.id, post.id)
+                resultComments = queriedPost.comments
+                resultBlog = queriedPost.blog
+                getPostCompleted.fulfill()
+            case .failure(let response):
+                XCTFail("Failed with: \(response)")
+            }
+        }
+        wait(for: [getPostCompleted], timeout: TestCommonConstants.networkTimeout)
+
+        guard let fetchedBlog = resultBlog else {
+            XCTFail("Could not get blog")
+            return
+        }
+
+        guard let fetchedComments = resultComments else {
+            XCTFail("Could not get comments")
+            return
+        }
+
+        // check if blog was eagerly loaded
+        XCTAssertEqual(fetchedBlog.id, blog.id)
```

Comment:
with above, then this becomes
```
guard let eagerlyLoadedBlog = fetchedPost.blog else {
  XCTFail("Blog is nil, should be loaded")
  return
}
```
The message in the XCTFail would also communicate enough info about the expectation that the comment can be removed